### PR TITLE
Metrics generators sample

### PR DESCRIPTION
**Pull request by omsoftdev**
This PR adds a sample of adding application metrics with the help of code generators.

### CODE REVIEW

File: eng/Versions.props [hunk 1]
```diff
@@ -40,6 +41,7 @@
     <MicrosoftExtensionsTelemetryAbstractionsVersion>8.1.0-preview.23564.2</MicrosoftExtensionsTelemetryAbstractionsVersion>
     <MicrosoftExtensionsTelemetryVersion>8.1.0-preview.23564.2</MicrosoftExtensionsTelemetryVersion>
     <MicrosoftExtensionsHttpDiagnosticsVersion>8.1.0-preview.23564.2</MicrosoftExtensionsHttpDiagnosticsVersion>
+    <MicrosoftExtensionsAuditReportsVersion>8.1.0-preview.23564.2</MicrosoftExtensionsAuditReportsVersion>
```
Comment (RussKie) [inline eng/Versions.props:44]: This needs to be noted in Version.Details.xml too.

Comment (xakep139) [reply eng/Versions.props:44]: @omsoftdev FYI

Comment (xakep139) [reply eng/Versions.props:44]: Oh, Igor fixed that in #172

File: src/Telemetry/Metering/Metrics.Generators/Model/RequestInfo.cs [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+// The structure that contains the metering information about the request.
+internal struct RequestInfo
+{
+    // This annotated property will be used as a tag for the RequestStats histogram.
+    [TagName(Tags.Target)]
+    public RequestTarget Target { get; set; }
+
+    // This annotated property will be used as a tag for the RequestStats histogram.
+    [TagName(Tags.DayOfWeek)]
+    public DayOfWeek DayOfWeek { get; set; }
```
[the remark this answers is not in the document]
Comment (xakep139) [reply src/Telemetry/Metering/Metrics.Generators/Model/RequestInfo.cs:19]: Addressed in #211

File: src/Telemetry/Metering/Metrics.Generators/Services/ConsoleMetricWriter.cs [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    private readonly MeterListener _meterListener;
+
+    public ConsoleMetricWriter()
+    {
+        _meterListener = new()
+        {
+            InstrumentPublished = (instrument, listener) =>
+            {
+                listener.SetMeasurementEventCallback<long>(PrintMeasurement);
+                listener.EnableMeasurementEvents(instrument);
```
[the remark this answers is not in the document]
Comment (xakep139) [reply src/Telemetry/Metering/Metrics.Generators/Services/ConsoleMetricWriter.cs:23]: Addressed in #211

File: src/Telemetry/Metering/Metrics.Generators/Model/RequestTarget.cs [hunk 4]
```diff
@@ -0,0 +1,12 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace Metrics.Generators;
+
+internal enum RequestTarget
+{
+    MicrosoftDotCom,
+    GitHubDotCom,
+    LinkedInDotCom,
```
[the remark this answers is not in the document]
Comment (xakep139) [reply src/Telemetry/Metering/Metrics.Generators/Model/RequestTarget.cs:10]: Addressed in #211

**Pull request merged**